Playerclass: Create 2022 demo

vanolson opened the pull request:
Jake, Katie, and I have created the 2022 demo. It was based in part on last year's demo, since we are displaying prefab classes and multiclasses (this time with prefabs). I added a function to display the details of our prefab skill trees, which we use for both the regular prefabs and multiclass prefabs. Katie created the new multiclass files and contributed to the helper functions, and wrote the multiclass_prefabs.h file. Jake did the bulk of the multiclass_prefabs.c file as he previously worked heavily on the multiclass module. Overall, this allows us to display the prefab skill trees we created this quarter, as well as our work on the multiclass module and new multiclass prefabs.

vanolson committed:
Duplicated 2021 demo, will modify for 2022

vanolson committed:
Removed unnecessary functions from the 2022 demo

Commit:
#1545 fixed some print statement for the new multiclass type. Also included header files from new modules.

vanolson committed:
Updated CMakeList to include 2022 demo

vanolson committed:
Merge branch 'playerclass-demo-2022' of github.com:uchicago-cs/chiventure into playerclass-demo-2022

vanolson committed:
Added null return for not-found class in class_prefabs, might break things

vanolson committed:
Fixed small formatting error in skilltrees demo

vanolson committed:
Fully added the 2022 demo to CMakeLists, should work properly now

vanolson committed:
Merge branch 'playerclass-demo-2022' of github.com:uchicago-cs/chiventure into playerclass-demo-2022

vanolson committed:
Added print skilltree function

vanolson committed:
Fixed skilltree print function

vanolson committed:
Updated skill display with skill type, added NULL cases

Commit:
pushing multiclass_prefab files

vanolson committed:
Added multiclass_prefabs.c to the CMake

vanolson committed:
Fixed reference to multiclass prefabs include file

Commit:
changes to see if chiventure builds

Commit:
second function

vanolson committed:
Created prefab functions in multiclass_prefabs.c

Commit:
Merge branch 'dev' of github.com:uchicago-cs/chiventure into playerclass-demo-2022

Commit:
some changes to multiclass_prefabs

Commit:
moved basic class into NPC specific section

Commit:
made some updates to the TO DO's - let me know if this is the right track

Commit:
added Hexblade prefab

Commit:
fixed merge conflicts

Commit:
finished multiclass prefabs. Now I have to add skills

Commit:
Merge branch 'dev' into playerclass-demo-2022

Commit:
just have to add skills then I'm done with multiclass prefabs

Commit:
nooo

Commit:
Merge branch 'playerclass-demo-2022' of github.com:uchicago-cs/chiventure into playerclass-demo-2022

Commit:
attempting multiclass_prefab_add_skills

Commit:
commented out the work i did bc its wrong so it doenst ness w work other people did

Commit:
good

Commit:
merge conflict

Commit:
fixed a bug

Commit:
skilltrees work? NO SKILLTREES

Commit:
hi from katies computer mine died. I have completed the skilltrees for multiclasses

Commit:
demmoo

Vincent-Zhang4135 commented on include/playerclass/class_prefabs.h:
```diff
@@ -47,4 +47,21 @@ class_t* class_prefab_new(game_t* game, char* class_name);
+
+//code in here is for integration purposes with multiclass_prefabs
+//and does not actually provide use outside of its own module 
+//see class_prefabs.c for more information
```
I appreciate the comment here. It definitely helped me understand the purpose of including it here as just for integrating with multiclass_prefabs. However, I think you can change the styling to use 


/*
 *
 */

instead

Vincent-Zhang4135 commented on include/playerclass/multiclass_prefabs.h:
Phenomenal documentation of function functionality

Vincent-Zhang4135 commented on src/playerclass/examples/2022-demo.c:
```diff
@@ -0,0 +1,296 @@
+    else {
+        printf("Skill Tree: NULL\n");
+    }
+
```
This function is very well structure. and is super self explanatory so great job! I guess to nitpick, you had a block comment for skills and names, which I don't think is entirely is necessary, but if you want to stay consistent, i think i could also add one for Description, Type, and (and then possible minimum XP/maximum levle)

vanolson replied:
Adding comments for the rest of them

Vincent-Zhang4135 commented on src/playerclass/examples/2022-demo.c:
```diff
@@ -0,0 +1,296 @@
+}
+
+
+/* A helper function for printing a class. */
```
Even if it seems unnecessary since it is literally taking in one parameter and not returning anything, I think it is still important important to follow the guide on commenting functions, so including the parameters, the expected return (nothing), and notes if there are any notes to consider

Vincent-Zhang4135 commented on src/playerclass/examples/2022-demo.c:
```diff
@@ -0,0 +1,296 @@
+#include "wdl/load_game.h"
+
+
+/* A helper function for printing a skilltree. */
```
Same thing here! (Even if it seems unnecessary since it is literally taking in one parameter and not returning anything, I think it is still important important to follow the guide on commenting functions, so including the parameters, the expected return (nothing), and notes if there are any notes to consider)

Vincent-Zhang4135 commented on src/playerclass/examples/2022-demo.c:
```diff
@@ -0,0 +1,296 @@
+}
+
+/* This path expects that we are at the root of the build directory */
+#define CLASSES_WDL_PATH "../tests/wdl/examples/wdl/classes.wdl"
```
I think it would be conventional to keep globally definite variables on the top of the file, even if you are using it way later in the file

Vincent-Zhang4135 commented on src/playerclass/examples/2022-demo.c:
```diff
@@ -0,0 +1,296 @@
+    strcat(zip_name, "./");
+    strcat(zip_name, "zip_default.zip");
+
+    // Create the zip
```
small nitpicking, but i think the comment should be moved inline if you do //. or if it is a block on its own line, be sure to following C styling guidelines of using /* */!

Vincent-Zhang4135 commented on src/playerclass/examples/2022-demo.c:
```diff
@@ -0,0 +1,296 @@
+    int error = 0;
+    zip_t *zip = zip_open(zip_name, ZIP_CREATE | ZIP_TRUNCATE, &error);
+
+    // Add DEFAULT.json to the zip
```
same thing here! (small nitpicking, but i think the comment should be moved inline if you do //. or if it is a block on its own line, be sure to following C styling guidelines of using /* */!)

Vincent-Zhang4135 commented on src/playerclass/examples/2022-demo.c:
```diff
@@ -0,0 +1,296 @@
+
+    zip_int64_t idx = zip_file_add(zip, data_name, zip_src, ZIP_FL_ENC_UTF_8);
+
+    // Write and save to disk
```
(same thing here)

Vincent-Zhang4135 commented on src/playerclass/examples/2022-demo.c:
```diff
@@ -0,0 +1,296 @@
+    int open_status;
+    zip = zip_open(zip_name, 0, &open_status);
+
+    // Read the zip into an obj
```
and here haha

Vincent-Zhang4135 commented on src/playerclass/examples/CMakeLists.txt:
```diff
@@ -5,7 +5,7 @@ add_library(playerclass_examples
 target_include_directories(playerclass_examples PRIVATE src/)
 
-set(PLAYERCLASS_EXAMPLES pc_demo 2021-demo)
+set(PLAYERCLASS_EXAMPLES pc_demo 2021-demo 2022-demo)
```
i am a little confused as I do not know what set does, but is there a reason why we set PLAYERCLASS EXAMPLES to both 2021-demo and 2022-demo? Instead of replacing 2021-demo with 2022-demo?

vanolson replied:
I suppose we could have deleted them, but the team last year kept in the 2020 demo (pc_demo) so we decided to just add ours rather than delete theirs. It still generates the old files I think, which isn't super necessary, but was helpful for us in drawing from last year's demo.

Vincent-Zhang4135 commented on src/playerclass/src/multiclass_prefabs.c:
```diff
@@ -0,0 +1,277 @@
+/* Skill related functions */
+
+/* 
+ * Skills are in a weird place right now: The skill trees team is interested in
```
I appreciate this comment, as it will inform futures devs of where we can further improve skill trees integration

Vincent-Zhang4135 commented on src/playerclass/src/multiclass_prefabs.c:
```diff
@@ -0,0 +1,277 @@
+        return EXIT_FAILURE;
+    }
+    return SUCCESS;
+}
```
This file in particular was really well written, understanding what each function does was really streamlined due to good comments and documentation.

Vincent-Zhang4135 reviewed (changes requested):
Great Job! After reviewing through all the files, I just had some styling suggestions (comments styling, function comments, etc.), and so I have decided to request for these small changes before moving onto to approving.

vanolson committed:
Implemented suggested changes for prefabs file and demo

Vincent-Zhang4135 reviewed (approved):
LGTM! The styling is more consistent with the actual style guide, and I will be passing this onto Lily for a final approval now! Good job to everyone involved creating a great PR

lilyehsani commented on src/playerclass/examples/2022-demo.c:
```diff
@@ -0,0 +1,314 @@
+}
+
+/* Helper function appropriated from WDL tests to load in a file */
+static obj_t *get_doc_obj()
```
Why do we need this function? I don't see it called anywhere.

lilyehsani commented on src/playerclass/examples/2022-demo.c:
```diff
@@ -0,0 +1,314 @@
+}
+
+
+/* Due to having to locate the example WDL file, it is critical that this executable 
```
What WDL file is being used? I don't see one in this PR.

lilyehsani reviewed (changes requested):
Looking good! Just a couple of comments.

vanolson committed:
Removed old code that was from last year's demo

vanolson committed:
Fixed merge conflict in multiclass header file

lilyehsani committed:
Merge branch 'dev' into playerclass-demo-2022

lilyehsani reviewed (approved):
LGTM! Approved.

lilyehsani committed:
Merge branch 'dev' into playerclass-demo-2022

vanolson commented:
Jake, Katie, and I created the 2022 demo. I created a function to display skilltrees and relevant information on their skills. Katie created the `multiclass_prefabs` header and C files, wrote `multiclass_prefabs.h` and added to the helper functions in the module. Jake wrote most of the `multiclass_prefabs.c` file, which expanded in part on `class_prefabs.c`. Vincent suggested a number of mainly style changes in his review, which I fixed.  In addition, Lily noticed old code that was still in the demo module, which I removed. The demo is now merged and ready to be used.

lilyehsani commented:
Issue Score: Excellent

Comments:
Great work on this PR!

<!-- {"score": "excellent", "minor": false} -->
